DOCS-960 Datadog IoT Agent (#7820)
DataDog/documentation: content/en/agent/iot/_index.md

- [Systemd][4]
- [NTP][5]

The IoT Agent also supports:

- Custom metric collection using an embedded [DogStatsD][6] server
- Log collection using [tailing files][7], [TCP/UDP][8], and [journald][9]

The IoT Agent does not include the Python interpreter and other integrations pre-packaged with the standard Agent. It also doesn’t support tracing for APM, live process monitoring, or network performance monitoring.

## Setup

### Requirements

The IoT Agent is available as DEB and RPM packages for Linux devices running on x64, arm64 (ARMv8), and ARMv7 architectures.

#### Resources

IoT devices are typically more resource constrained than cloud infrastructure hosts. The IoT Agent is built to have a minimal footprint and consume minimal network bandwidth.

Exact resource requirements depend on usage. Datadog found the following when testing the IoT Agent (v7.20) internally:

- CPU: 0.5% (on a VM with 2 Intel Xeon VCPUs)
- Memory: 36 MB
- Network bandwidth: 237 bps up / 79 bps down
- Disk: 63 MB

### Installation

#### Automatic

To automatically download and install the correct IoT Agent for your operating system and chipset architecture, use the following command:

```shell

> DD_API_KEY=<YOUR_DD_API_KEY> DD_AGENT_MAJOR_VERSION=7 DD_AGENT_FLAVOR=iot-agent bash -c "$(curl -L https://raw.githubusercontent.com/DataDog/datadog-agent/master/cmd/agent/install_script.sh)"

Reviewer: Noticed an error while testing this. Should be: DATADOG_AGENT_FLAVOR=datadog-iot-agent

Reviewer 2: Should be `DD_AGENT_FLAVOR=datadog-iot-agent`, and we are changing the URL for that script also, let me fix it.

> DD_API_KEY=<YOUR_DD_API_KEY> DD_AGENT_MAJOR_VERSION=7 DD_AGENT_FLAVOR=iot-agent bash -c "$(curl -L https://raw.githubusercontent.com/DataDog/datadog-agent/master/cmd/agent/install_script.sh)"

Reviewer 2: ```suggestion
DD_API_KEY=<YOUR_DD_API_KEY> DD_AGENT_MAJOR_VERSION=7 DD_AGENT_FLAVOR=datadog-iot-agent bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script.sh)"
```

```

#### Manual

{{< tabs >}}
{{% tab "DEB" %}}

To manually install the IoT Agent on Debian-based operating systems, run the following commands:

> 1. Update `apt` and install `apt-transport-https` to download through https:

Reviewer 3: ```suggestion
1. Update `apt` and install `apt-transport-https` to download through HTTPS:
```

    sudo apt-get install datadog-iot-agent
    ```

4. Copy the example config and plug in your API key:
    ```bash
    DD_API_KEY=<YOUR_DD_API_KEY> ; sudo sh -c "sed 's/api_key:.*/api_key:$DD_API_KEY/' /etc/datadog-agent/datadog.yaml.example > /etc/datadog-agent/datadog.yaml"
    ```

5. Start the IoT Agent:
    ```bash
    sudo systemctl restart datadog-agent.service
    ```

{{% /tab %}}
{{% tab "RPM" %}}

To manually install the IoT Agent on RPM-based operating systems, run the following commands:

1. Set up Datadog's Yum repo on your system by creating `/etc/yum.repos.d/datadog.repo` with the contents:
    ```
    [datadog]
    name = Datadog, Inc.
    baseurl = https://yum.datadoghq.com/stable/7/<HOST_OS>
    enabled=1
    gpgcheck=1
    gpgkey=https://yum.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public
    ```
    
    The `baseurl` is dependent on your host OS:
    - x86_64 - `https://yum.datadoghq.com/stable/7/x86_64/`
    - arm64 - `https://yum.datadoghq.com/stable/7/aarch64/`
    - ARMv7 - `https://yum.datadoghq.com/stable/7/armv7hl/`

2. Update your local yum repo and install the Agent:
    ```
    sudo yum makecache
    sudo yum install datadog-iot-agent
    ```

> 3. Copy the example config and plug in your API key

Reviewer 3: ```suggestion
3. Copy the example config and plug in your API key:
```

    ```bash
    DD_API_KEY=<YOUR KEY> ; sudo sh -c "sed 's/api_key:.*/api_key:$DD_API_KEY/' /etc/datadog-agent/datadog.yaml.example > /etc/datadog-agent/datadog.yaml"
    ```

> 

Author: @albertvaka - We have a partial that allows us to display information based on which region the customer has selected on the docs site, for example:

```
{{< site-region region="us" >}}

DD_API_KEY=<YOUR KEY> ; sudo sh -c "sed 's/api_key:.*/api_key:$DD_API_KEY/' /etc/datadog-agent/datadog.yaml.example > /etc/datadog-agent/datadog.yaml"

{{< /site-region >}}
{{< site-region region="eu" >}}

<EU_COMMAND_HERE>

{{< /site-region >}}
```

What would be the EU command?

> 

Reviewer 2: For automatic installation, pass `DD_SITE` to the install_script. Install command for EU becomes:

```
DD_API_KEY=<YOUR_DD_API_KEY> DD_SITE=datadoghq.eu DD_AGENT_MAJOR_VERSION=7 DD_AGENT_FLAVOR=datadog-iot-agent bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script.sh)"
```

For manual installation, this additional step is needed:

```
sudo sh -c "sed 's/# site:.*/site: datadoghq.eu/' /etc/datadog-agent/datadog.yaml > /etc/datadog-agent/datadog.yaml.new && mv /etc/datadog-agent/datadog.yaml.new /etc/datadog-agent/datadog.yaml
```

As a side note, though, this selector won't scale as we add support for new regions... ideally we would have something like this:

```
DD_SITE={{< get-site-region-url >}} [...]
```

Which would work on all regions.

> 

Author: Thanks! I will discuss with the websites team.

---
title: IoT Agent
kind: documentation
beta: true
further_reading:
  - link: "/getting_started/agent/"
    tag: "Documentation"
    text: "Getting Started with the Agent"
---

## Overview

The Datadog IoT Agent is a version of the Agent optimized for monitoring IoT devices and embedded applications. Customers use the IoT Agent to monitor a wide variety of devices from digital displays to security devices running image detection algorithms.

<div class="alert alert-warning">

> The IoT Agent is currently available for private beta. Request access by completing the <a href="https://www.datadoghq.com/">form</a>.

Author: @jimmycaputo - I will need this from you before we merge.

> The IoT Agent is currently available for private beta. Request access by completing the <a href="https://www.datadoghq.com/">form</a>.

Reviewer: Here's the Google Form link: https://docs.google.com/forms/d/e/1FAIpQLSfaoqr9Pgievq7f2WE6wcQMF1YjsOwiXHbmO2FehUlQwmNu0A/viewform?usp=sf_link

</div>

## Capabilities

> The IoT Agent includes the following system checks. Configuration is identical to the full Agent.

Reviewer 2: I'm not a big fan of using "full Agent" in public-facing docs. The official name of the product is "Datadog Agent", but this might not be enough to distinguish it from the "Datadog IoT Agent"... I don't know how to solve this problem, though. I can only think of other adjectives like "regular Datadog Agent" or "main Datadog Agent", which I'm not sure are better.

Checking what others do, Facebook has a "Facebook Lite" product and they don't call the full app "full Facebook": https://www.facebook.com/help/1278187755533783

Author: I agree and also not sure how to handle this. @jimmycaputo

Reviewer: It doesn't bother me that much but maybe a better way to make this point is: Configuration for IoT devices is identical to other types of hosts.

Reviewer 2: Indeed, rewording can solve the problem, but note there are 3 occurrences of "full Agent" in this document, so we will need to reword each of them differently.

Author: I re-worded this one as Jimmy suggested and changed "full Agent" to "standard Agent" in the other places.

- [System][1] (includes CPU, IO, load, memory, swap, and uptime)
- [Disk][2]
- [Network][3]
- [Systemd][4]
- [NTP][5]

The IoT Agent also supports:

- Custom metric collection using an embedded [DogStatsD][6] server
- Log collection using [tailing files][7], [TCP/UDP][8], and [journald][9]

The IoT Agent does not include the Python interpreter and other integrations pre-packaged with the full Agent. It also doesn’t support tracing for APM, live process monitoring, or network performance monitoring.

## Setup

### Requirements

> The IoT Agent is available on Debian and RPM for both 64-bit x86 and Arm v8 devices.

Reviewer: Would be better to say Debian and RPM-based systems I think.

> The IoT Agent is available on Debian and RPM for both 64-bit x86 and Arm v8 devices.

Author: ```suggestion
The IoT Agent is available on Debian and RPM-based systems for both 64-bit x86 and Arm v8 devices.
```

> The IoT Agent is available on Debian and RPM-based systems for both 64-bit x86 and Arm v8 devices.

Reviewer 2: ```suggestion
The IoT Agent is available as DEB and RPM packages for Linux devices running on x64, arm64 (ARMv8) and ARMv7 architectures.
```
- Explicitly mention Linux is the only supported OS.
- IoT Agent 7.21 will be available for ARMv7 already.

#### Resources

IoT devices are typically more resource constrained than cloud infrastructure hosts. The IoT Agent is built to have a minimal footprint and consume minimal network bandwidth.

Exact resource requirements depend on usage. Datadog found the following when testing the IoT Agent (v7.20) internally:

- CPU: 0.5% (on a VM with 2 Intel Xeon VCPUs)
- Memory: 36 MB
- Network bandwidth: 237 bps up / 79 bps down
- Disk: 63 MB

### Installation

#### Automatic

To automatically download and install the correct IoT Agent for your operating system and chipset architecture, use the following command:

```shell
DD_API_KEY=<YOUR_DD_API_KEY> DD_AGENT_MAJOR_VERSION=7 DD_AGENT_FLAVOR=iot-agent bash -c "$(curl -L https://raw.githubusercontent.com/DataDog/datadog-agent/master/cmd/agent/install_script.sh)"
```

#### Manual

{{< tabs >}}

> {{% tab "Debian" %}}

Reviewer 2: ```suggestion
{{% tab "DEB" %}}
```

    sudo apt-get update
    sudo apt-get install datadog-iot-agent
    ```

4. Copy the example config and plug in your API key:
    ```bash
    DD_API_KEY=<YOUR_DD_API_KEY> ; sudo sh -c "sed 's/api_key:.*/api_key:$DD_API_KEY/' /etc/datadog-agent/datadog.yaml.example > /etc/datadog-agent/datadog.yaml"
    ```

5. Start the IoT Agent:
    ```bash
    sudo systemctl restart datadog-agent.service
    ```

{{% /tab %}}
{{% tab "RPM" %}}

To manually install the IoT Agent on RPM-based operating systems, run the following commands:

1. On an x86_64 host, set up Datadog's Yum repo on your system by creating `/etc/yum.repos.d/datadog.repo` with the contents:
    ```
    [datadog]
    name = Datadog, Inc.
    baseurl = https://yum.datadoghq.com/stable/7/x86_64/
    enabled=1
    gpgcheck=1
    gpgkey=https://yum.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public
    ```

2. On an arm64 host, set up Datadog's Yum repo on your system by creating `/etc/yum.repos.d/datadog.repo` with the contents:
    ```
    [datadog]
    name = Datadog, Inc.
    baseurl = https://yum.datadoghq.com/stable/7/aarch64/
    enabled=1
    gpgcheck=1
    gpgkey=https://yum.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public

>     ```

Reviewer 2: For ARMv7 the URL would be:
```
    baseurl = https://yum.datadoghq.com/stable/7/armv7hl/
```

    sudo apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 A2923DFF56EDA6E76E55E492D3A80E30382E94DE
    ```

3. Update `apt` and install the IoT Agent:
    ```bash
    sudo apt-get update
    sudo apt-get install datadog-iot-agent
    ```

{{< site-region region="us" >}}

<ol start="4">
  <li><p>Copy the example config and plug in your API key:</p>
  <pre><code>DD_API_KEY=&lt;YOUR_DD_API_KEY&gt; ; sudo sh -c "sed 's/api_key:.*/api_key:$DD_API_KEY/' /etc/datadog-agent/datadog.yaml.example > /etc/datadog-agent/datadog.yaml"</code></pre>
  </li>
</ol>

{{< /site-region >}}
{{< site-region region="eu" >}}

<ol start="4">
  <li><p>Copy the example config and plug in your API key:</p>
  <pre><code>DD_API_KEY=&lt;YOUR_DD_API_KEY&gt; ; sudo sh -c "sed 's/api_key:.*/api_key:$DD_API_KEY/' /etc/datadog-agent/datadog.yaml.example > /etc/datadog-agent/datadog.yaml"</code></pre>
  <li><p>Set your Datadog site:</p>
  <pre><code>sudo sh -c "sed 's/# site:.*/site: datadoghq.eu/' /etc/datadog-agent/datadog.yaml > /etc/datadog-agent/datadog.yaml.new && mv /etc/datadog-agent/datadog.yaml.new /etc/datadog-agent/datadog.yaml</code></pre>
  </li>
  </li>
</ol>

{{< /site-region >}}

> <ol start="6">

Reviewer 2: ```suggestion
<ol start="5">
```

Author: There's 2 bullet points above this one (4 and 5) so this group needs to start with `6`.

To manually install the IoT Agent on RPM-based operating systems, run the following commands:

1. Set up Datadog's Yum repo on your system by creating `/etc/yum.repos.d/datadog.repo` with the contents:
    ```
    [datadog]
    name = Datadog, Inc.

>     baseurl = https://yum.datadoghq.com/stable/7/<HOST_OS>

Reviewer 2: ```suggestion
    baseurl = https://yum.datadoghq.com/stable/7/<HOST_ARCHITECTURE>
```
